dsl/src/testground-runner.ts

```diff
@@ -0,0 +1,83 @@
+import { tmpdir } from 'os'
+import { promises as fs } from 'fs';
+import * as dsl from './dsl'
+import { transform } from './testground-transform'
+import path from 'path';
+import { execSync, spawn } from 'child_process';
+import { env } from 'process';
+
+export async function run(testplans: dsl.TestPlans) {
+    const dir = path.join(tmpdir(), "testground")
+
+    // Check if directory exists
+    try {
+        await fs.access(dir)
+        await fs.rm(dir, { recursive: true, force: true })
+    } catch (e) {
+    }
+    await fs.mkdir(dir)
+    await fs.mkdir(path.join(dir, "testplans"))
+
+    const filesToCreate = transform(testplans)
```

Comment:
Within `transform` we assume that all containers have been built already (which they are currently via the Makefile). What do you think of moving the building of the images also into here?

I am asking because a next step for this could be to pull images instead of building them but that would probably be a bit clunky to express in the Makefile so it might be better done in here.

Reply:
https://github.com/testground/testground/issues/1522 is what I am aiming at!

Reply:
I think it's best if we don't try to build here. I think that's a sizable bit of complexity in testground. The Makefile is just one strategy to build the container image. Using Nix can be another. Maybe we want to use [warpforge](https://github.com/warpfork/warpforge) at some point.

It's nice that this just has to reference some existing image id, and then it'll work.